api/v1alpha1/clusterconfig_types.go

```diff
@@ -344,6 +348,66 @@ func (ImageRegistries) VariableSchema() clusterv1.VariableSchema {
 	}
 }
 
+type Users []User
+
+func (Users) VariableSchema() clusterv1.VariableSchema {
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Description: "Users to add to the machine",
+			Type:        "array",
+			Items:       ptr.To(User{}.VariableSchema().OpenAPIV3Schema),
+		},
+	}
+}
+
+// User defines the input for a generated user in cloud-init.
+type User struct {
+	// Name specifies the user name
+	Name string `json:"name"`
+
+	// Passwd specifies a hashed password for the user
+	// +optional
+	Passwd *string `json:"passwd,omitempty"`
```

Comment:
Is this is a hashed password, can we call it `hashedPassword` or `passwordHash`?
```suggestion
	HashedPasswordd *string `json:"hashedPassword,omitempty"`
```

Reply:
I started by using the field names in the CABPK types, but I agree, `HashedPassword` is an improvement.